src/server/utils/bucket_diff.js

```diff
@@ -165,9 +168,12 @@ class BucketDiff {
      * @returns {nb.BucketDiffKeysDiff}
      */
     _object_grouped_by_key_and_omitted(list) {
-        if (!list) return {};
+        if (!list || (this.for_deletion && !this.version)) return {};
```

Comment:
Let's add a comment here.
also, maybe we want to return an error on this setup (deletion + non version)

Reply:
Sure, will do this.